src/engine/authentication.md

```diff
@@ -0,0 +1,51 @@
+# Authentication
+
+The `engine` JSON-RPC interface, exposed by EL and consumed by CL, needs to be authenticated. The authentication scheme chosen for thus purpose is [JWT](https://jwt.io/). 
+
+The type of attacks that this authentication scheme attempts to protect against are the following: 
+
+- RPC port exposed towards the internet, allowing attackers to exchange messages with EL engine api. 
+- RPC port exposed towards the browser, allowing malicious webpages to submit messages to the EL engine api. 
+
+The authentication scheme is _not_ designed to 
+
+- Prevent attackers with capability to read ('sniff') network traffic from reading the traffic, 
+- Prevent attackers with capability to read ('sniff') network traffic from performing replay-attacks of earlier messages. 
+
+Authentication is performed as follows: 
+
+- For `HTTP` dialogue, each `jsonrpc` request is individually authenticated by supplying `JWT` token in the HTTP header. 
+- For a WebSocket dialogue, only the initial handshake is authenticated, after which the message dialogue proceeds without further use of JWT.
+  - Clarification: The websocket handshake starts with the client performing a websocket upgrade request. This is a regular http GET request, and the actual
+parameters for the WS-handshake is carried in the http headers. 
+- For `inproc`, a.k.a raw ipc communication, no authentication is required, under the assumption that a process able to access `ipc` channels for the process, which usually means local file access, is already sufficiently permissioned that further authentication requirements do not add security. 
+
+
+## JWT specifications
+
+- The authenticated `engine` api **MUST** use a different port from the legacy API. The EL **SHOULD** default to using port `8546`.
```

Comment:
This statement is in the engine api today:

> Client software MUST expose Engine API at a port independent from JSON-RPC API. The default port for the Engine API is 8550. The Engine API is exposed under the engine namespace.

1. this is a different suggested port
2. the core of the engine API can/should probably just link to this doc instead of repeating the info

Reply:
I suggest that we switch it to port 8551. If we do that, EL's can easily roll out both, and have them on both 8550 and 8551 for a period of time until all EL/CL have implemented it, and EL can start shutting down `8550`

Reply:
I see. Different one for authenticated

Given we have a breaking change coming to the core of the engine API to be released alongside this authentication, do you think it worth handling the backwards compatibility by having both ports open? Instead we could use this as an opportunity to just make it required outright, given the new testnets coming.

Or do you see value in not blocking interop testing on the engine API otherwise between now and then?